### PR TITLE
BPB-188: So many alerts

### PR DESCRIPTION
**Pull request by scribe**
I suspect we could clean it up more, this is a first stab.

### CODE REVIEW

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/createbucket/CreateBucketPresenter.java [hunk 1]
```diff
@@ -145,19 +145,19 @@ public void createBucket() {
                     });
                 });
                 createBucketTask.setOnFailed(event -> {
-                    alert.showAlert(resourceBundle.getString("createBucketErrorAlert"));
+                    alert.showAlert(resourceBundle.getString("createBucketErrorAlert"), "Error");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/createbucket/CreateBucketPresenter.java]: Shouldn't we be getting the `"Error"` text from the `resourceBundle` as well?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java [hunk 2]
```diff
@@ -428,22 +429,22 @@ private Session getSession() {
...
+            errorAlert.showAlert(resourceBundle.getString("invalidSession"), "Error");
             return;
         }
 
         // Verify valid remote TreeTableView
         final TreeTableView<Ds3TreeTableValue> ds3TreeTableView = getTreeTableView();
         if (ds3TreeTableView == null) {
             LOG.info("Files not selected");
-            alert.showAlert(resourceBundle.getString("fileSelect"));
+            infoAlert.showAlert(resourceBundle.getString("fileSelect"), "Info" );
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: Shouldn't we be getting the `"Info"` string from the resource bundle?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java [hunk 3]
```diff
@@ -447,7 +447,7 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
             });
             jobWorkers.execute(putJob);
         } else {
-            alert.showAlert(resourceBundle.getString("operationNotAllowedHere"));
+            warnAlert.showAlert(resourceBundle.getString("operationNotAllowedHere"), "Warning");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java]: Same question about the "Warning" text and the resourceBundle.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/newsession/NewSessionPresenter.java [hunk 4]
```diff
@@ -188,20 +189,20 @@ public void cancelSession() {
...
         } else {
             if (Guard.isNotNullAndNotEmpty(ds3SessionStore.getObservableList())) {
                 ds3SessionStore.getObservableList().forEach(openSession -> {
                     if (savedSessions.getSelectionModel().getSelectedItem().getName().equals(openSession.getSessionName())) {
-                        alert.showAlert(resourceBundle.getString("cannotdeletesession"));
+                        infoAlert.showAlert(resourceBundle.getString("cannotdeletesession"), "Information");
                     } else {
                         savedSessionStore.removeSession(savedSessions.getSelectionModel().getSelectedItem());
-                        alert.showAlert(resourceBundle.getString("sessionDeletedSuccess"));
+                        infoAlert.showAlert(resourceBundle.getString("sessionDeletedSuccess"), "Information");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/newsession/NewSessionPresenter.java]: And similar question about resourceBundle.

**Comment by scribe** [comment]
I think I fixed it, but it's a big change so please give LazyAlert another look.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 5]
```diff
@@ -16,49 +16,66 @@
...
-            final Stage stage = (Stage) alert.getDialogPane().getScene().getWindow();
-            stage.getIcons().add(new Image(ImageURLs.DEEP_STORAGE_BROWSER));
+        }
+        alert.setTitle(title);
+        alert.setHeaderText(null);
 
+        final DialogPane dp = alert.getDialogPane();
+        if(dp != null) {
+            final Scene scene = dp.getScene();
+            if(scene !=null) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Space on the right side of the `!=` is missing.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 6]
```diff
@@ -16,49 +16,66 @@
...
+               final Stage w = (Stage) scene.getWindow();
+               if(w != null) {
+                   w.getIcons().add(new Image( ImageURLs.DEEP_STORAGE_BROWSER));
+               }
+            }
         }
+
         alert.setContentText(message);
         alert.showAndWait();
+        alert.setAlertType(alertType);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: This should be set before the `showAndWait` call

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 7]
```diff
@@ -16,49 +16,66 @@
...
     }
 
-    public LazyAlert(final String title) {
-        this(title, Alert.AlertType.INFORMATION);
-    }
 
-    private void showAlertInternal(final String message) {
-        if (alert == null) {
+    private void showAlertInternal(final String message, final String title, final Alert.AlertType alertType) {
+        if(alert == null) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Missing space after `if`

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 8]
```diff
@@ -16,49 +16,66 @@
...
-            alert.setHeaderText(null);
-
-            final Stage stage = (Stage) alert.getDialogPane().getScene().getWindow();
-            stage.getIcons().add(new Image(ImageURLs.DEEP_STORAGE_BROWSER));
+        }
+        alert.setTitle(title);
+        alert.setHeaderText(null);
 
+        final DialogPane dp = alert.getDialogPane();
+        if(dp != null) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Missing space after `if`

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 9]
```diff
@@ -16,49 +16,66 @@
...
+            final Scene scene = dp.getScene();
+            if(scene !=null) {
+               final Stage w = (Stage) scene.getWindow();
+               if(w != null) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Missing space after `if`

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java (hunk 5 above)
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Missing space after `if`

**Pull request merged**